### PR TITLE
Core/Movement: Add MovePoint with facing

### PR DESCRIPTION
**Pull request by Kittnz**
Thanks to Riztazz

**Changes proposed:**

-  Core/MotionMaster: Add MovePoint with facing

**Target branch(es):** 3.3.5

- [x] 3.3.5

**Tests performed:** (Does it build, tested in-game, etc.)
Works as is.

Notes: This is needed for Children of Goldshire event #20745

### CODE REVIEW

File: src/server/game/Movement/MotionMaster.h [hunk 1]
```diff
@@ -129,6 +129,7 @@ class TC_GAME_API MotionMaster
             MovePoint(id, pos.m_positionX, pos.m_positionY, pos.m_positionZ, generatePath);
         }
         void MovePoint(uint32 id, float x, float y, float z, bool generatePath = true);
+        void MovePoint(uint32 id, Position const& pos, bool generatePath, bool _setFinalFacing);
```
Comment (Keader) [inline src/server/game/Movement/MotionMaster.h]: setFinalFacing

File: src/server/game/Movement/MotionMaster.cpp [hunk 2]
```diff
@@ -339,6 +339,21 @@ void MotionMaster::MovePoint(uint32 id, float x, float y, float z, bool generate
     }
 }
 
+void MotionMaster::MovePoint(uint32 id, Position const& pos, bool generatePath = true, bool _setFinalFacing = false)
```
Comment (Keader) [inline src/server/game/Movement/MotionMaster.cpp]: setFinalFacing

**Comment by tkrokli** [comment]
About the title edit: I simply picked the directory name 1 level up
(which also is the common denominator for all files in this PR).

**Review by Keader — APPROVED** [review]
Looks good to me

**Comment by ccrs** [comment]
doesnt feel right.. everything should be just handled with Position and update all the x,y,z uses..
stall this for the weekend, maybe I can perform the changes

**Comment by Riztazz** [comment]
Afaik shauren didnt want to use NAN to determine if there is orientation set, because by default orient is set to some value, 0.0f i think? Even tho 0.0f is still valid orient ;p

**Comment by Keader** [comment]
Maybe has some useful discussion about it [here](https://github.com/TrinityCore/TrinityCore/pull/18893)

**Comment by Shauren** [comment]
#include "Optional.h"

Use that instead of float+bool arg pair (dont shove it into Position)

**Comment by jackpoz** [comment]
@Kittnz can you use the suggestion above about using Optional ? Then this PR could be merged

**Comment by Kittnz** [comment]
Ill look into it.

**Review by Naios — APPROVED** [review]

**Review by Naios — CHANGES_REQUESTED** [review]
As said before, the optional part is missing

File: src/server/game/Movement/MovementGenerators/PointMovementGenerator.cpp [hunk 3]
```diff
@@ -50,6 +50,10 @@ void PointMovementGenerator<T>::DoInitialize(T* owner)
     init.MoveTo(_x, _y, _z , _generatePath);
     if (_speed > 0.0f)
         init.SetVelocity(_speed);
+
+    if (_orient)
+        init.SetFacing(*_orient);
```
Comment (jackpoz) [inline src/server/game/Movement/MovementGenerators/PointMovementGenerator.cpp]: are these 2 lines correct ?

**Comment by jackpoz** [comment]
@Naios could you check if it's ok now ?

**Review by Shauren — CHANGES_REQUESTED** [review]
No, Optional was meant to propagate in api back to MotionMaster::MovePoint, with the overload that has Position argument not being added

File: src/server/game/Movement/MovementGenerators/PointMovementGenerator.cpp (hunk 3 above)
Comment (Naios) [reply src/server/game/Movement/MovementGenerators/PointMovementGenerator.cpp]: Yes, this should be correct

**Review by Naios — APPROVED** [review]
Optional used correctly instead of 0 float,
Please address Shaurens comment before merging please

**Comment by jackpoz** [comment]
@Shauren better now ? (I actually deleted the overload before reading your comment since it didn't make sense to add a new one to me either)

**Comment by Shauren** [comment]
`Optional<float> = nullptr` only works because its we defined `BOOST_OPTIONAL_USE_OLD_DEFINITION_OF_NONE` to work around msvc linker errors, you should default construct it with `Optional<float> = {}`

**Review by Shauren — APPROVED** [review]

**Pull request merged**